resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientApplication.scala

```diff
@@ -159,6 +159,19 @@ private[spark] class Client(
         logInfo(s"Waiting for application $appName to finish...")
         watcher.awaitCompletion()
         logInfo(s"Application $appName finished.")
+        try {
```

Comment:
Why are we doing this in client code? Driver shutdown is the right place to perform cleanup right?

Reply:
(also because this code path isn't invoked in fire-and-forget mode IIUC)

Reply:
We talked about this before. The main reason is this requires giving the driver extra permissions to delete things.

Reply:
I see! It was the RBAC rules and downscoping them that led us here. I'm concerned not all jobs will actually use this interactive mode of launching. What do you think of just granting more permissions to the driver and allowing cleanup there?

Reply:
Per the discussion offline, I think the right solution is move resource management to the driver pod. This way, resource cleanup is guaranteed regardless of the deployment mode.